### PR TITLE
[CAIM-54] Add dynamic fields conditioning in foster application

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Better usability and validation for foster form.

## Change

- adds script tag with additional form logic

## Ticket

https://caimcommunity.atlassian.net/browse/CAIM-54

### CODE REVIEW

File: caim_base/templates/fosterer_profile/edit.html [hunk 1]
```diff
@@ -132,4 +132,60 @@ <h3>Your Fosterer Profile: Page {{ stageNumber }} of 6</h3>
...
+            // Initialize for Pets and People in Home
+            updateDetailsVisibility('[name="num_existing_pets"]', '.pet-detail');
+            updateDetailsVisibility('[name="num_people_in_home"]', '.person-detail');
+
+            // Make rent specific fields disappear for homeowners
+            var rentRestrictionsInput =  document.querySelector('#div_id_rent_restrictions');
+            var landlordContactText =  document.querySelector('#div_id_landlord_contact_text');
+
+            function toggleRentRestrictions() {
+                console.log('sdsd');
```
Comment (rgasper) [inline caim_base/templates/fosterer_profile/edit.html]:
```suggestion
```
:)

**Review by rgasper — APPROVED** [review]
I tested it out and it's working as advertised. Validated that the page works and that the data is in the database.

Not needed to be part of this PR, but:
Is it possible to put a `DeprecationWarrning` on accessing a model field? It'd be nice to do that for the old `reference_1`, `reference_2`, `reference_3` and the `people_in_home_detail` fields that have been replaced by FK to another model. That threw me for a loop for a few minutes.

File: caim_base/templates/fosterer_profile/edit.html (hunk 1 above)
Comment (Johnetordoff) [reply caim_base/templates/fosterer_profile/edit.html]: :) WIP!

File: caim_base/models/fosterer.py [hunk 2]
```diff
@@ -304,12 +304,21 @@ class RentOwn(models.TextChoices):
     reference_1 = models.TextField(
         blank=True, null=True, default=None, verbose_name="Reference #1"
     )
+    reference_1.system_check_deprecated_details = dict(
```
Comment (rgasper) [inline caim_base/models/fosterer.py:307]: Sick. this is why I love django

**Pull request merged**